metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -573,6 +573,17 @@ class MetalsLanguageServer(
     }
   }
 
+  @JsonNotification("metals/windowStateDidChange")
+  def windowStateDidChane(params: WindowStateDidChangeParams): Unit = {
+    CompletableFuture.completedFuture {
+      if(params.focused) {
+        compileSourceFiles.unpause
```

Comment:
We should also unpause/pause the cascade compile queue.

Reply:
And I've also added pausing on changed sbt build files.